src/utils.js

```diff
@@ -33,7 +33,13 @@ async function sendCompat(ethereum, method, params) {
   if (ethereum.sendAsync && ethereum.selectedAddress) {
     return new Promise((resolve, reject) => {
       ethereum.sendAsync(
-        { method, params, from: ethereum.selectedAddress },
+        {
+          method,
+          params,
+          from: ethereum.selectedAddress,
+          jsonrpc: '2.0',
+          id: 0,
```

Comment:
Interesting—do you know if we need to keep track of this and increment it with each new request?

Reply:
It seems like that should only be necessary for batch requests: see the spec here https://www.jsonrpc.org/specification

Given it's only used for ad-hoc single requests I think it's ok to leave as 0 (but apparently it ought to be a number rather than null).